conans/test/functional/graph_lock/dynamic_test.py

```diff
@@ -403,18 +403,20 @@ def configure(self):
     def _check(self):
         client = self.client
 
-        def _validate():
-            client.run("lock create conanfile.py --name=LibD --version=1.0 --lockfile=libb.lock "
-                       "--lockfile-out=libd.lock", assert_error=True)
-            expected = ("LibA/1.0: LibC/1.0 tried to change LibA/1.0 option myoption to False\n"
-                        "but it was already defined as True")
-            self.assertIn(expected, client.out)
-
-        client.save({"conanfile.py": GenConanfile().with_require("LibB/1.0")
-                                                   .with_require("LibC/1.0")})
-        _validate()
+        client.save({"conanfile.py": GenConanfile().with_requires("LibB/1.0", "LibC/1.0")})
+        client.run("lock create conanfile.py --name=LibD --version=1.0 --lockfile=libb.lock "
+                   "--lockfile-out=libd.lock", assert_error=True)
+        expected = textwrap.dedent("""\
+                       ERROR: LibA/1.0: Locked options do not match computed options
+                       Locked options:
+                       myoption=False
+                       Computed options:
+                       myoption=True""")
+        self.assertIn(expected, client.out)
+
+        # Order of LibC, LibB does matter, in this case it will not raise
+        client.save({"conanfile.py": GenConanfile().with_requires("LibC/1.0", "LibB/1.0")})
```

Comment:
This test has changed, it cannot raise an error anymore. It will be a new (missing) binary for the added ``LibC``, that will get the locked option for ``LibA:myoption``. Depending on the evaluation error it can be different, and this is annoying but at least it is correct in the sense that it respects the lockfile, but in some cases it will error early, and in other cases the locked value will prevail and if anything it can fail later (missing binary, or binary cannot be built).

This is not ideal, but I think it is complicated and it is not worth such a huge investment, and this can wait until the new graph in Conan 2.0